src/services/decoratemobiledrawinteraction.js

```diff
@@ -0,0 +1,93 @@
+goog.provide('ngeo.DecorateMobileDrawInteraction');
+
+goog.require('goog.asserts');
+goog.require('ngeo');
+
+
+/**
+ * Provides a function that adds the following properties (using
+ * `Object.defineProperty`) to a mobile draw interaction, making it possible
+ * to watch those properties:
+ *
+ * - "dirty"
+ * - "drawing"
+ * - "valid"
+ *
+ * Note that only getters are provided.
+ *
+ * @example
+ * <div ngModel="interaction.drawing">Drawing</div>
+ *
+ * @typedef {function(ngeo.interaction.MobileDraw)}
+ * @ngdoc service
+ * @ngname ngeoDecorateInteraction
+ */
+ngeo.DecorateMobileDrawInteraction;
+
+
+/**
+ * @param {angular.Scope} $rootScope
+ * @return {ngeo.DecorateMobileDrawInteraction}
+ */
+ngeo.decorateMobileDrawInteraction = function($rootScope) {
+  return (
+
+      /**
+        * @param {ngeo.interaction.MobileDraw} interaction Interaction
+        *     to decorate.
+        */
+      function(interaction) {
+        goog.asserts.assertInstanceof(interaction, ngeo.interaction.MobileDraw);
+
+        // NOTE: the 'dirty' property is manually managed (see below).
+        // The reason is this: 'dirty' is set to `true` by the interaction
+        // while angular is out of scope, which means that it doesn't detect
+        // it being changed. To work around this, the change event is listened
+        // and the property is manually changed
+        //
+        // A better approach to avoid doing this would be great, as this is
+        // not pretty at all...
+        Object.defineProperty(interaction, 'dirty', {
+          writable: true,
+          value: false
+        });
+
+        Object.defineProperty(interaction, 'drawing', {
+          get: function() {
+            return interaction.getDrawing();
+          }
+        });
+        Object.defineProperty(interaction, 'valid', {
+          get: function() {
+            return interaction.getValid();
+          }
+        });
+
+
+        goog.events.listen(
+            interaction,
+            ol.Object.getChangeEventType(
+                ngeo.interaction.MobileDrawProperty.DIRTY),
+            function() {
+              var dirty = interaction.getDirty();
+              Object.defineProperty(interaction, 'dirty', {
+                value: dirty
+              });
+
+              // this is where the angular scope is forced to be applied. We
+              // only need to do this when dirty, as going to "no being dirty"
+              // is made by a click on a button where Angular is within scope
+              if (dirty) {
+                $rootScope.$apply();
```

Comment:
For all ol3 event, controller code is responsible of calling the `scope.$apply() method`.
Here, the directive that will manage the state of the button, depending on dirty value, will do the apply.